lib/policy_machine.rb

```diff
@@ -195,6 +195,20 @@ def scoped_privileges(user_or_attribute, object_or_attribute, options = {})
   end
 
   ##
+  # Search for and iterate over a collection in batches
+  def batch_find(type:, query: {}, config: {}, &blk)
+    if policy_machine_storage_adapter.respond_to?(:batch_find)
+      policy_machine_storage_adapter.batch_find(type, query, config, &blk)
+    else
+      method(type.to_s.pluralize).call.select do |pm_el|
```

Comment:
Would `method(type.to_s.pluralize)`.call(query)` work instead of the select? Also, this looks like it doesn't handle the block, and assumes a batch size of 1. I think something like `.each_slice(config[:batch_size] || 1, &blk)` would work, it'll still return an enumerator if there is no block.